### PR TITLE
Readme rewrite

### PR DESCRIPTION
**Pull request by efriis**

### CODE REVIEW

File: templates/rag-redis/README.md [hunk 1]
```diff
@@ -21,57 +13,59 @@ export REDIS_USER = <YOUR REDIS USER NAME>
...
 
-| Environment Variable | Description                       | Default Value |
-|----------------------|-----------------------------------|---------------|
-| `DEBUG`            | Enable or disable Langchain debugging logs       | True         |
-| `REDIS_HOST`           | Hostname for the Redis server     | "localhost"   |
-| `REDIS_PORT`           | Port for the Redis server         | 6379          |
-| `REDIS_USER`           | User for the Redis server         | "" |
-| `REDIS_PASSWORD`       | Password for the Redis server     | "" |
-| `REDIS_URL`            | Full URL for connecting to Redis  | `None`, Constructed from user, password, host, and port if not provided |
-| `INDEX_NAME`           | Name of the vector index          | "rag-redis"   |
```
Comment (efriis) [inline templates/rag-redis/README.md:58]: transfer

File: templates/csv-agent/README.md [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+
+If you want to add this to an existing project, you can just run:
+
+```shell
+langchain app add csv-agent
+```
+
+And add the following code to your `server.py` file:
+```python
+from csv_agent.agent import chain as csv_agent.agent_chain
```
Comment (efriis) [inline templates/csv-agent/README.md:32]: fix

File: templates/elastic-query-generator/README.md [hunk 3]
```diff
@@ -1,31 +1,82 @@
...
+
+If you want to add this to an existing project, you can just run:
+
+```shell
+langchain app add elastic-query-generator
+```
+
+And add the following code to your `server.py` file:
+```python
+from elastic_query_generator.chain import chain as elastic_query_generator.chain_chain
```
Comment (efriis) [inline templates/elastic-query-generator/README.md:48]: fix

File: templates/neo4j-cypher/README.md [hunk 4]
```diff
@@ -25,14 +14,75 @@ NEO4J_USERNAME=<YOUR_NEO4J_USERNAME>
...
+
+We can access the template from code with:
+
+```python
+from langserve.client import RemoteRunnable
+
+runnable = RemoteRunnable("http://localhost:8000/neo4j_cypher")
+```
+
+## Neo4j database setup
```
Comment (efriis) [inline templates/neo4j-cypher/README.md:75]: probably better above langserve instructions

File: templates/neo4j-generation/README.md [hunk 5]
```diff
@@ -1,44 +1,75 @@
...
 
-There are a number of ways to set up a Neo4j database.
+## Usage
 
-### Neo4j Aura
+To use this package, you should first have the LangChain CLI installed:
 
-Neo4j AuraDB is a fully managed cloud graph database service.
-Create a free instance on [Neo4j Aura](https://neo4j.com/cloud/platform/aura-graph-database?utm_source=langchain&utm_content=langserve).
-When you initiate a free database instance, you'll receive credentials to access the database.
```
Comment (efriis) [inline templates/neo4j-generation/README.md:24]: need to add this back

File: templates/plate-chain/README.md [hunk 6]
```diff
@@ -1 +1,61 @@
-# plate-chain
```
Comment (efriis) [inline templates/plate-chain/README.md:1]: need a title

File: templates/plate-chain/README.md [hunk 7]
```diff
@@ -1 +1,61 @@
...
+If you wish to add this to an existing project, simply run:
+
+```shell
+langchain app add plate-chain
+```
+
+Then add the following code to your `server.py` file:
+
+```python
+from plate_chain import chain as plate_chain_chain
```
Comment (efriis) [inline templates/plate-chain/README.md:30]: `plate_chain_chain` is weird

File: templates/rag-elasticsearch/README.md [hunk 8]
```diff
@@ -1,59 +1,85 @@
...
+
+```shell
+langchain app add rag-elasticsearch
+```
+
+And add the following code to your `server.py` file:
+```python
+from rag-elasticsearch import chain as rag-elasticsearch_chain
+
+add_routes(app, rag-elasticsearch_chain, path="/rag-elasticsearch")
```
Comment (efriis) [inline templates/rag-elasticsearch/README.md:45]: fix

File: templates/rag-fusion/README.md [hunk 9]
```diff
@@ -1,5 +1,66 @@
-# RAG Fusion
+# rag-fusion
 
-Re-implemented from [this GitHub repo](https://github.com/Raudaschl/rag-fusion), all credit to original author
+RAG Fusion is a re-implementation of the project found at [this GitHub repo](https://github.com/Raudaschl/rag-fusion). All credit goes to the original author. It's a search methodology that aims to bridge the gap between traditional search paradigms and the multifaceted dimensions of human queries. Inspired by the capabilities of Retrieval Augmented Generation (RAG), this project goes a step further by employing multiple query generation and Reciprocal Rank Fusion to re-rank search results.
```
Comment (efriis) [inline templates/rag-fusion/README.md:3]: do we want the indented quoteblock back?

File: templates/rag-semi-structured/README.md [hunk 10]
```diff
@@ -1,47 +1,73 @@
...
+
+We can access the template from code with:
+
+```python
+from langserve.client import RemoteRunnable
+
+runnable = RemoteRunnable("http://localhost:8000/rag-semi-structured")
+```
+
+For more details on how to connect to the template, refer to the Jupyter notebook `rag_semi_structured`.
```
Comment (efriis) [inline templates/rag-semi-structured/README.md:73]: should we link this

File: templates/rag-supabase/README.md
Comment (efriis) [inline templates/rag-supabase/README.md:1]: this one dropped a ton of stuff

File: templates/rag-weaviate/README.md [hunk 11]
```diff
@@ -1,16 +1,71 @@
-# RAG Weaviate
 
-This template performs RAG using Weaviate and OpenAI.
+# rag-weaviate
 
-##  Weaviate
+This package performs RAG (Retriever-Augmented Generation) using Weaviate and OpenAI.
```
Comment (efriis) [inline templates/rag-weaviate/README.md:4]: Retrieval

File: templates/self-query-supabase/README.md
Comment (efriis) [inline templates/self-query-supabase/README.md:1]: also dropped a ton

File: templates/sql-ollama/README.md [hunk 12]
```diff
@@ -1,18 +1,74 @@
...
+If you want to add this to an existing project, you can just run:
+
+```shell
+langchain app add sql-ollama
+```
+
+And add the following code to your `server.py` file:
+
+```python
+# TODO: Add appropriate app route code
```
Comment (efriis) [inline templates/sql-ollama/README.md:41]: todo

Comment (efriis) [reply templates/sql-ollama/README.md:41]: also check pyproject for issue

File: templates/xml-agent/README.md [hunk 13]
```diff
@@ -1,17 +1,70 @@
...
 
-This template will use `Anthropic` by default. 
+## Environment Setup
 
-Be sure that `ANTHROPIC_API_KEY` is set in your environment.
+Two environment variables need to be set:
 
-##  Tools
+- `ANTHROPIC_API_KEY`: Required for using Anthropic
+- `YDC_API_KEY`: Required for using You.com
```
Comment (efriis) [inline templates/xml-agent/README.md:11]: remove - I think this is duckduckgo now

File: templates/pirate-speak/README.md [hunk 14]
```diff
@@ -1,9 +1,69 @@
...
-##  LLM
+```python
+from langserve.client import RemoteRunnable
 
-This template will use `OpenAI` by default. 
+runnable = RemoteRunnable("http://localhost:8000/pirate-speak")
+```
 
-Be sure that `OPENAI_API_KEY` is set in your environment.
+---
```
Comment (efriis) [inline templates/pirate-speak/README.md:69]: remove

File: templates/rag-pinecone-rerank/README.md [hunk 15]
```diff
@@ -1,23 +1,73 @@
...
+## Environment Setup
 
-This connects to a hosted Pinecone vectorstore.
-
-Be sure that you have set a few env variables in `chain.py`:
+Ensure the following environment variables are set in `chain.py`:
 
 * `PINECONE_API_KEY`
 * `PINECONE_ENV`
 * `index_name`
```
Comment (efriis) [inline templates/rag-pinecone-rerank/README.md:12]: lowercase?

File: templates/rag-pinecone-rerank/README.md [hunk 16]
```diff
@@ -1,23 +1,73 @@
...
-[Re-ranking](https://docs.cohere.com/docs/reranking) provides a way to rank retrieved documents using specified filters or criteria.
+This package performs RAG using Pinecone and OpenAI, with Cohere to perform re-ranking on returned documents. Re-ranking provides a way to rank retrieved documents using specified filters or criteria.
 
-##  Pinecone
+## Environment Setup
 
-This connects to a hosted Pinecone vectorstore.
-
-Be sure that you have set a few env variables in `chain.py`:
+Ensure the following environment variables are set in `chain.py`:
```
Comment (efriis) [inline templates/rag-pinecone-rerank/README.md:8]: is this chain.py or environment?

File: templates/anthropic-iterative-search/README.md [hunk 17]
```diff
@@ -1,3 +1,69 @@
+
```
Comment (hwchase17) [inline templates/anthropic-iterative-search/README.md:1]: ?

Comment (hwchase17) [inline templates/anthropic-iterative-search/README.md:1]:
```suggestion
```

File: templates/hyde/README.md [hunk 18]
```diff
@@ -1,9 +1,76 @@
...
-Hypothetical Document Embeddings (HyDE) are a method to improve retrieval.
-To do this, a hypothetical document is generated for an incoming query.
-That document is then embedded, and that embedding is used to look up real documents similar to that hypothetical document.
-The idea behind this is that the hypothetical document may be closer in the embedding space than the query.
-For a more detailed description, read the full paper [here](https://arxiv.org/abs/2212.10496).
+# hyde
+
+This template HyDE with RAG. 
+
+Hyde is a Python package that stands for Hypothetical Document Embeddings (HyDE). It is a method used to enhance retrieval by generating a hypothetical document for an incoming query.
```
Comment (hwchase17) [inline templates/hyde/README.md:6]:
```suggestion
Hyde is a retrieval method that stands for Hypothetical Document Embeddings (HyDE). It is a method used to enhance retrieval by generating a hypothetical document for an incoming query. 
```

File: templates/neo4j-generation/README.md (hunk 5 above)
Comment (rlancemartin) [reply templates/neo4j-generation/README.md]: done

File: templates/plate-chain/README.md (hunk 6 above)
Comment (rlancemartin) [reply templates/plate-chain/README.md:1]: done.

File: templates/plate-chain/README.md (hunk 7 above)
Comment (rlancemartin) [reply templates/plate-chain/README.md:37]: done.

File: templates/rag-fusion/README.md (hunk 9 above)
Comment (rlancemartin) [reply templates/rag-fusion/README.md]: cleaned this.

File: templates/rag-semi-structured/README.md (hunk 10 above)
Comment (rlancemartin) [reply templates/rag-semi-structured/README.md:75]: done.

File: templates/sql-ollama/README.md (hunk 12 above)
Comment (rlancemartin) [reply templates/sql-ollama/README.md]: removed.

File: templates/xml-agent/README.md (hunk 13 above)
Comment (rlancemartin) [reply templates/xml-agent/README.md]: done, not there now.

File: templates/rag-pinecone-rerank/README.md (hunk 15 above)
Comment (rlancemartin) [reply templates/rag-pinecone-rerank/README.md]: fixed.

File: templates/rag-pinecone-rerank/README.md (hunk 16 above)
Comment (rlancemartin) [reply templates/rag-pinecone-rerank/README.md]: fixed.

**Pull request merged**